lib/cubit/meal_logic.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import 'package:mobile_boilerplate_flutter/screens/meal_detail_screen.dart';
+import 'package:mobile_boilerplate_flutter/screens/meal_selection_screen.dart';
+import 'package:mobile_boilerplate_flutter/screens/welcome_screen.dart';
+
+import 'meal_cubit.dart';
+
+class MealLogic extends StatefulWidget {
+  const MealLogic({Key? key}) : super(key: key);
+
+  @override
+  _MealLogicState createState() => _MealLogicState();
+}
+
+class _MealLogicState extends State<MealLogic> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(body: BlocBuilder<MealCubit, MealState>(
+      builder: (context, state) {
+        if (state is MealInitial) {
+          return const WelcomeScreen();
+        }
+        if (state is MealLoading) {
```

Comment:
The Loading part became tricky in the end, when I realised I had to make the app more responsive. Solution is not perfect, but I learned a lot.